GeneralisedFilters/src/containers.jl

```diff
@@ -8,39 +8,69 @@
 A container for particle filters which composes the weighted sample into a distibution-like
 object, with the states (or particles) distributed accoring to their log-weights.
 """
-mutable struct ParticleDistribution{PT,WT<:Real}
+abstract type ParticleDistribution{PT} end
+
+Base.collect(state::PT) where {PT<:ParticleDistribution} = state.particles
```

Comment:
```suggestion
Base.collect(state::ParticleDistribution) = state.particles
```
I think this is equivalent. Same for several below.

Also, since `ParticleDistribution` is an abstract type, and here we are getting a literal field of it called `.particles`, would be good to document that all subtypes of `ParticleDistribution` must have such a field. Or, maybe even better, wrap it in a `getparticles` function, and then document that all subtypes must implement `getparticles`.